New font files

GHaberis opened the pull request:
## What does this change?

**This has been updated for the reasons outlined in my [comment](https://github.com/guardian/frontend/pull/20466#issuecomment-427415598) below **

1. Replace all font assets in `frontend/static/src/fonts/` with new font files for:

- GuardianHeadline (noalts-not-hinted and noalt-autohinted)
- GuardianTextEgyptianWeb (noalts-not-hinted and noalt-autohinted)
- GuardianSansWeb (noalts-not-hinted and noalt-autohinted)

2. In `tools/__tasks__/compile/fonts/index.config.js` update config so new font assets are encoded and the correct font-weights are used for the matching font files.

3. Updates `common/app/templates/inlineJS/blocking/loadFonts.scala.js` to remove logic for deciding whether to use `cleartype` hinting as the new fonts only have `auto` hinting variants.

I've tested on CODE/Browserstack and only found a couple places where the weight was noticeably different with the next fonts (eg. Pillar links in the nav). I've updated the font-weight in these occurences. I've spoken with @blongden73 and @zeftilldeath about the risk that some elements may get a different weight and they've agreed it's OK to release and fix these as and when we become aware of them. 

## TODO

- Remove "kerning-off" logic, kerning-off variants for fonts have never been used and are no longer available, yet we have logic scattered throughout the code for supporting "kerning-off".

- Update the hardcoded font-loading logic in [error.scala.html](https://github.com/guardian/frontend/blob/master/identity/app/views/errors/error.scala.html) and [notFound.scala.html](https://github.com/guardian/frontend/blob/master/archive/app/views/notFound.scala.html) as the localStorage fallback expects [webfonts-hinting-cleartype-kerning-on.scss](https://github.com/guardian/frontend/blob/master/static/src/stylesheets/webfonts-hinting-cleartype-kerning-on.scss) to exist. Once we've updated these files we can delete this css file. Once deleted we can update 
[javaScriptConfig.scala.js](https://github.com/guardian/frontend/blob/master/common/app/templates/javaScriptConfig.scala.js) to remove the `cleartype` option in `window.guardian.config.stylesheets.fonts` and also remove the unnecessary `kerningOn` nesting.

## Screenshots

## What is the value of this and can you measure success?

The new files are more performant and they contain wider character sets. 

## Checklist

### Does this affect other platforms?

- AMP - This doesn't change the CSS or font files used on our AMP templates.
- Email Templates - This doesn't change the CSS or font files used on our AMP templates.

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

The following keys used by [commercial templates](https://github.com/guardian/commercial-templates/blob/master/src/_shared/js/messages.js#L69-L74
) will need updating and font-family names in the CSS will also need to updated.

### Does this change break ad-free?

No

PRBuilds commented:
PRbuilds results:


**Screenshots**
[wide.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/wide.png) • [desktop.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/desktop.png) • [tablet.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/tablet.png) • [mobile.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/mobile.png)



&#x1F49A; **Exceptions**
[thrown-exceptions.js](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/thrown-exceptions.js)





&#x1F525; (failed)  **A11y validation**
[a11y-report.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/a11y-report.txt)



&#x1F49A; **Microdata Validation**
[microdata.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/microdata.txt)

    

**Apache Benchmark Load Testing**
[loadtesting.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/loadtesting.txt)



**LightHouse Reporting**
[lighthouse.html](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-20466/lighthouse.html)


    
--automated message

GHaberis commented:
@paperboyo thanks for your comment, points 1 and 2 were simply down to typo in the PR (I've updated the text in the original comment). Regarding point 3 that's great to know. 

> these were, admittedly arbitrarily, restrained to sparser subsets for less important styles because of the weight. I strongly advice following this logic for everything but woff2 now too. So:

- For `woff2` we should use the `noalts` subset across all weights? 
- For `woff` and `ttf` we should use the `latin1` subsets? Is this across all weights? Or should we use the `noalts` set for the 'regular' variant and `latin1` for the _italic_ and **bold** variants?

katebee commented:
@GHaberis CommDev work might take a while...

![shockcat](https://user-images.githubusercontent.com/8607683/46491198-f540fe80-c801-11e8-993f-a79dc9812f76.gif)

paperboyo commented on tools/__tasks__/compile/fonts/index.config.js:
```diff
@@ -1,435 +1,507 @@
+        'font-family': '"Guardian Text Egyptian"',
         fonts: [
             {
                 'font-weight': '300',
```
Regular weight should always be 400, I think. 300 is for Light. Also weight numbers are missing for Regular weights above for all flavours but ttf hinted.

paperboyo commented on tools/__tasks__/compile/fonts/index.config.js:
```diff
@@ -1,435 +1,507 @@
         fonts: [
             {
-                src: 'GuardianTextSans/noalts-not-hinted/GuardianTextSans-Regular.woff2'
+                src:
```
Font weight numbers similarly missing? (maybe this will fix TextSans Bold replacing Regular on CODE?)

paperboyo commented on tools/__tasks__/compile/fonts/index.config.js:
```diff
@@ -1,973 +1,507 @@
+        'font-family': '"Guardian Text Egyptian"',
         fonts: [
             {
                 src:
```
Regular weight should always be 400, I think. 300 is for Light. Also weight numbers are missing for Regular weights above for all flavours but ttf hinted.

paperboyo commented on tools/__tasks__/compile/fonts/index.config.js:
```diff
@@ -1,973 +1,507 @@
+        'font-family': '"Guardian Text Egyptian"',
         fonts: [
             {
+                'font-weight': '300',
```
Regular weight should always be 400, I think. 300 is for Light. Also weight numbers are missing for Regular weights above for all flavours but ttf hinted.

paperboyo commented on tools/__tasks__/compile/fonts/index.config.js:
```diff
@@ -346,18 +346,18 @@ module.exports = [
             {
                 'font-weight': '300',
```
Regular weight should always be 400, I think. 300 is for Light. Also weight numbers are missing for Regular weights above for all flavours but ttf hinted.

GHaberis commented:
> I can’t even leave a comment so that it’s not outdated

I think I get the gist of the comments, ok...

So reagrding missing font-weights I followed the existing `index.config.js` as guidance. In that file  `font-weight` is not explicitly defined for `Regular` variants of `GuardianTextSansWeb` and `GuardianTextEgyptianWeb`. I'm happy to explicitly associate Regular with `'font-weight': '400'`.

Anywhere you've seen `'font-weight': '300'` mapping to `Regular` variants is a mistake which I'll correct (map to `'font-weight': '300'`) now!

paperboyo commented:
> In that file `font-weight` is not explicitly defined for `Regular` variants of `GuardianTextSansWeb` and `GuardianTextEgyptianWeb`. I'm happy to explicitly associate `Regular` variants with `'font-weight': '400'`.

I honestly don’t know if that’s needed. If any kind of resetting or a notion of “main weight” is necessary/useful for dotcom or other entities using these… E.g. I have no idea if these foundry-provided files (“fonts.css”) added in this PR are even used by anything. If it isn’t – superb!

> Anywhere you've seen `'font-weight': '300'` mapping to `Regular` variants is a mistake which I'll correct (map to `'font-weight': '400'`) now!

Thanks! Just this one place, I think.

GHaberis commented:
> I honestly don’t know if that’s needed. If any kind of resetting or a notion of “main weight” is necessary/useful for dotcom or other entities using these… 

I've tested and I don't believe it's needed. Without explicitly mapping `font-weight: 400` to `Regular` you get `Regular` on any `font-weight` < 600, after that with `font-weight` >= 600 you get `Bold`. After  explicitly mapping `font-weight: 400` to `Regular` I see exactly the same behaviour. 

> E.g. I have no idea if these foundry-provided files (“fonts.css”) added in this PR are even used by anything. If it isn’t – superb!

They're not!

paperboyo commented:
Not sure if it belongs to this PR (whatever’s easiest), but we’ve just discovered that one style of GH Guardian Headline is only used in one-and-a-half places on the site and Design decided to remove it completely 🎆 :
GH Guardian Headline Semibold

I will make a case to fix some faux styles separately, but now the case is stronger 😉 .

And a little thing, I can see on CODE:
![image](https://user-images.githubusercontent.com/6032869/46533202-b069a600-c89b-11e8-8dab-b17531635c78.png)
Is there a reason the family name is missing `GH` at the beginning of it’s name?

GHaberis commented:
@paperboyo 

>  we’ve just discovered that one style of GH Guardian Headline is only used in one-and-a-half places on the site and Design decided to remove it completely

Does this mean I can remove `GHGuardianHeadline-Semibold` from `index.config.js`?

> Is there a reason the family name is missing `GH` at the beginning of it’s name?

No, I've just referred to the foundries `.css` and they've defined the family as "GH Guardian Headline" so for consistencies sake I'll update to do the same!

GHaberis commented:
> Are the current fonts going to vanish completely once this is merged? 

@katebee let's chat about this. The current fonts won't vanish from user's local storage after this PR is merged, if they have them they'll remain there so when DFP ads request them they'll be returned as they are now. The current fonts won't however be stored in localStorage for any new users. The only thing I'm unsure of is how the commercial templates behave if the fonts they request via postMessage aren't in local storage.

GHaberis commented:
Having looked into this it appears the renaming of the `localStorage` keys and the `font-family` names may cause commercial some disruption as they have hardcoded references to these keys/family names in code for each template uploaded to DFP. 

If we reuse the existing keys (`GuardianHeadline`/`GuardianTextEgyptianWeb`/`GuardianTextSansWeb`) and family-names (`Guardian Egyptian Web`/`Guardian Text Egyptian Web`/`Guardian Text Sans Web`) it should make the upgrade less disruptive for commercial, some font-weights on some templates may be off but that's it, they won't need to change the family-names in their css or the key references used for retrieving font styles.

I believe my original worry about needing to update the keys to invalidate the font styles already stored in local storage isn't a problem as the keys contain a hash which should update after this has been released (TBC by testing on CODE).

@katebee @paperboyo

paperboyo commented:
Whatever cause less disruption/work for friends in Commercial is good. Let’s just try to be (very, very slowly) prepared for calling them proper names once dotcomponents goodness arrives.

GHaberis commented:
@SiAdcock @paperboyo @katebee OK I've tested reusing the existing keys (`GuardianHeadline`/`GuardianTextEgyptianWeb`/`GuardianTextSansWeb`) and family-names (`Guardian Egyptian Web`/`Guardian Text Egyptian Web`/`Guardian Text Sans Web`) and it all works - so user's with the old font styles in their localStorage will have these removed and replaced with the new styles. And it won't require anywork from commercial!

GHaberis commented:
I've reverted the commit https://github.com/guardian/frontend/pull/20466/commits/32414a0db34884fef3643c50943c7faab1e4da74 where I deleted the old font files from the repo because of the 4000+ "files changed" by it meant you couldn't really view this PR in github! I'll remove these files in a separate PR.

mchv commented:
Excellent work @GHaberis 🎖
